src/wast-parser.y

```diff
@@ -665,9 +665,9 @@ expr1 :
       assert(if_->type == ExprType::If);
       if_->if_.true_->label = $2;
     }
-  | try_check LPAR BLOCK labeling_opt block RPAR catch_list {
-      $5->label = $4;
+  | try_check labeling_opt LPAR BLOCK block RPAR catch_list {
```

Comment:
Ok, was able to rewrite so no explicit (block ....) had to be added around the try body.